integration/tests/cook/test_impersonation.py

```diff
@@ -119,6 +119,9 @@ def test_self_impersonate(self):
         user1 = self.user_factory.new_user()
         job_uuids = []
         try:
+            with self.user_factory.admin():
```

Comment:
This change seems unrelated to the PR. Split off?

Reply:
Why is this code in this specific location, in this test? This looks to be setup/reset code that belongs somewhere else.

Reply:
Yep, this is unrelated to the default submit pool change. That being said, I saw failures for these tests on this branch, so it seems appropriate to fix them on this branch.

As for why it's here, the resetting needs to happen for the specific user the test is using, which is determined in the test.

Reply:
Thanks for explanation. 

Are the test failures caused by the intended code changes in this branch?

Reply:
Nope, they were caused by a test run that was aborted and left a test user's quota in a too-low state.

Reply:
*hmms* this still smells a bit off. Shouldn't we have something that resets the whole cluster to a known good state, if this can go wrong?

Reply:
Good call, moved these calls to the `UserFactory`